modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v3_2_4/UpgradeDDMContent.java

```diff
@@ -0,0 +1,207 @@
+
+			try (ResultSet rs = ps1.executeQuery()) {
+				while (rs.next()) {
+					String definition = rs.getString("definition");
+
+					JSONObject definitionJSONObject =
+						_jsonFactory.createJSONObject(definition);
+
+					String data = rs.getString("data_");
+
+					JSONObject dataJSONObject = _jsonFactory.createJSONObject(
+						data);
+
+					if (_upgradeDDMContentData(
+							dataJSONObject.getJSONArray("fieldValues"),
+							definitionJSONObject.getJSONArray("fields"))) {
+
+						ps2.setString(1, dataJSONObject.toJSONString());
+
+						long contentId = rs.getLong("contentId");
+
+						ps2.setLong(2, contentId);
+
+						ps2.addBatch();
+					}
+				}
+			}
+
+			ps2.executeBatch();
+		}
+	}
+
+	private DecimalFormat _getDecimalFormat(Locale locale) {
+		DecimalFormat decimalFormat = _decimalFormatsMap.get(locale);
+
+		if (decimalFormat == null) {
+			decimalFormat = (DecimalFormat)DecimalFormat.getInstance(locale);
+
+			decimalFormat.setGroupingUsed(false);
+			decimalFormat.setMaximumFractionDigits(Integer.MAX_VALUE);
+			decimalFormat.setParseBigDecimal(true);
+
+			_decimalFormatsMap.put(locale, decimalFormat);
+		}
+
+		return decimalFormat;
+	}
+
+	private HashMap<String, JSONObject> _mapDataFieldValues(
+		JSONArray fieldValuesJSONArray) {
+
+		HashMap<String, JSONObject> dataFieldValuesMap = new HashMap<>();
+
+		fieldValuesJSONArray.forEach(
+			object -> {
+				JSONObject fieldValueJSONObject = (JSONObject)object;
+
+				dataFieldValuesMap.put(
+					fieldValueJSONObject.getString("name"),
+					fieldValueJSONObject.getJSONObject("value"));
+			});
+
+		return dataFieldValuesMap;
+	}
+
+	private boolean _upgradeDDMContentData(
+		JSONArray fieldValuesJSONArray, JSONArray fieldsJSONArray) {
+
+		AtomicBoolean upgraded = new AtomicBoolean(false);
+
+		HashMap<String, JSONObject> dataFieldValuesMap = _mapDataFieldValues(
+			fieldValuesJSONArray);
+
+		fieldsJSONArray.forEach(
+			object -> {
+				JSONObject fieldJSONObject = (JSONObject)object;
+
+				String type = fieldJSONObject.getString("type");
+
+				if (type.equals("numeric")) {
+					String name = fieldJSONObject.getString("name");
+
+					JSONObject fieldValueJSONObject = dataFieldValuesMap.get(
+						name);
+
+					JSONArray namesJSONArray = fieldValueJSONObject.names();
+
+					namesJSONArray.forEach(
+						languageId -> {
+							try {
+								DecimalFormat decimalFormat = _getDecimalFormat(
+									LocaleUtil.fromLanguageId(
+										GetterUtil.getString(languageId)));
+
+								String valueString =
+									fieldValueJSONObject.getString(
+										GetterUtil.getString(languageId));
+
+								Number number = GetterUtil.getNumber(
+									decimalFormat.parse(valueString));
+
+								String formattedNumber = decimalFormat.format(
+									number);
+
+								if (!valueString.equals(formattedNumber)) {
+									DecimalFormat normalDecimalFormat =
+										_getDecimalFormat(LocaleUtil.US);
```

Comment:
No. I changed my mind about that solution because it would require a lot more changes and adaptations than just fixing the numbers that weren't being localised after evaluation. You're probably wondering about that fixed US locale. It is there because if the persisted number is not in their native format, it's in the US format, so I need the US `DecimalFormat` to parse it to a `Number` type.

Reply:
Oh I see, thanks for your confirmation, @rodrigopaulino. Can we add this locale to a variable, like `defaultDecimalFormatLocale`, for example, just to reinforce this information in the code? I understand that doing this will mitigate the chances of this information doesn't be comprehended by you, me or other developers in the future. 😉